santoku/utils/test/test_url_handler.py

```diff
@@ -1,209 +1,50 @@
-    # Each element in the list consists of: the raw dataframe, exploded dataframe if
-    # `raise_exception_if_invalid_url` is set to `True`, and the exploded dataframe if
-    # `raise_exception_if_invalid_url` is set to `False`
-    return [
-        pd.DataFrame(
-            {
-                "referrer": [
-                    "https://sub1.example.com/path?query#fragment",
-                    "http://sub2.sub1.example.com?query#fragment",
-                    "https://localhost.com",
-                    "https://example.co.uk",
-                    "https://example.com.uk",
-                    "www.example.com",
-                    "https://localhost.example",
-                    "localhost",
-                    "125.0.0.0",
-                    "com",
-                    "//",
-                    "",
-                ]
-            }
-        ),
-        pd.DataFrame(
-            {
-                "domain": [
-                    "example.com",
-                    "sub1.example.com",
-                    "example.com",
-                    "sub1.example.com",
-                    "sub2.sub1.example.com",
-                    "localhost.com",
-                    "example.co.uk",
-                    "com.uk",
-                    "example.com.uk",
-                    "example.com",
-                    "www.example.com",
-                ]
-            }
-        ),
-        pd.DataFrame(
-            {
-                "domain": [
-                    "example.com",
-                    "sub1.example.com",
-                    "example.com",
-                    "sub1.example.com",
-                    "sub2.sub1.example.com",
-                    "localhost.com",
-                    "example.co.uk",
-                    "com.uk",
-                    "example.com.uk",
-                    "example.com",
-                    "www.example.com",
-                    "example",
-                    "localhost.example",
-                    "localhost",
-                    "125.0.0.0",
-                    "com",
-                    "//",
-                    "",
-                ]
-            }
-        ),
-    ]
-
-
-def test_get_partial_domain(raw_urls_num_subdomains_and_partial_domain):
-    (
-        input_url,
-        num_subdomains,
-        reference_partial_domain_raising_exception_for_invalid_url,
-        reference_partial_domain_not_raising_exception_for_invalid_url,
-    ) = raw_urls_num_subdomains_and_partial_domain
+        ("https://localhost.example", 0, None),
+        ("https://sub1.localhost.example", 1, None),
+        ("https://sub1.localhost.example", 2, None),
+        ("localhost", 0, None),
+        ("https://125.0.0.0", 1, None),
+        ("com", 0, None),
+        ("*", 0, None),
+        ("//", 0, None),
+        ("", 0, None),
+        ("https:///integration/ckeditor", 0, None),
+    ],
+    scope="function",
+)
+def test_get_partial_domain_raising_exception_for_invalid_url(
```

Comment:
I remember discussing this test. This idea came to mind, does it look correct and more readable to you?
```python
    try:
        output_result = URLHandler.get_partial_domain(url=input_url, num_subdomains=num_subdomains, raise_exception_if_invalid_url=True)
        assert output_result == expected_result
    except e:
        with pytest.raises(InvalidURLError):
            raise e
```

Reply:
Alternatively, since there are two types of test cases and results, maybe a more elegant solution is to split this into two tests, one for cases were an exception is expected and one for a regular execution.